tests/federation_room_send_test.go

```diff
@@ -249,3 +253,70 @@ func TestNetworkPartitionOrdering(t *testing.T) {
 		},
 	})
 }
+
+func TestAllowInvalidTransactionPDU(t *testing.T) {
+	deployment := complement.Deploy(t, 1)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Register(t, "hs1", helpers.RegistrationOpts{})
+
+	srv := federation.NewServer(t, deployment,
+		federation.HandleKeyRequests(),
+		federation.HandleMakeSendJoinRequests(),
+		federation.HandleTransactionRequests(nil, nil),
+	)
+
+	cancel := srv.Listen()
+	defer cancel()
+
+	nexy := srv.UserID("nexy")
+
+	room := alice.MustCreateRoom(t, map[string]interface{}{
+		"preset": "public_chat",
+	})
+	server_room := srv.MustJoinRoom(t, deployment, "hs1", room, nexy)
+
+
+	event := srv.MustCreateEvent(t, server_room, federation.Event{
+		Sender: nexy,
+		Type: "m.room.message",
+		Content: map[string]interface{}{"body": "hello i am nexy and i somehow broke the room into a v1 room"},
+	})
+	//event.EventID()
+	data, err := json.Marshal(event)
+	if err != nil {
+		t.Log("invalid json bwuh", err)
+		t.FailNow()
+	}
+	data, err = sjson.SetBytes(data, "event_id", event.EventID())
+
+	client := srv.FederationClient(deployment)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+	resp, err := client.SendTransaction(ctx, gomatrixserverlib.Transaction{
+		TransactionID: gomatrixserverlib.TransactionID(fmt.Sprintf("complement-%d", time.Now().Nanosecond())),
+		Origin: spec.ServerName(srv.ServerName()),
+		Destination: "hs1",
+		PDUs: []json.RawMessage{data},
+		EDUs: nil,
+	})
+
+	if err != nil {
+		ct.Fatalf(t, "Invalid PDU in transaction made the request fail")
+	}
+
+	for eventID, e := range resp.PDUs {
+		if eventID != "$meow" && eventID != event.EventID() {
+			ct.Fatalf(t, "Server responded with bogus event ID %s", eventID)
+		} else if e.Error == "" {
+			ct.Fatalf(t, "Server accepted event sent into a V10 room using V1 format")
```

Comment:
At least for Dendrite, this is fine. When it receives a transaction, [it extracts the room ID, asks the DB about the version and then executes ](https://github.com/element-hq/dendrite/blob/7cc7ebb46f1f06d08747199542a164f0400e0b5e/internal/transactionrequest.go#L106-L135) `NewEventFromUntrustedJSON` ([link](https://github.com/element-hq/dendrite/blob/7cc7ebb46f1f06d08747199542a164f0400e0b5e/internal/transactionrequest.go#L106-L135)), which for a `V10` room (not specified in the test) runs `newEventFromUntrustedJSONV2` ([link](https://github.com/matrix-org/gomatrixserverlib/blob/dbd5f31fefc031633c3418165e4ef6d343e03999/eventV2.go#L122-L138)), which strips out the `event_id`.

Reply:
Added a bogus prev_event too, now it will get the INFO log about not including the event id in the error dict instead